pkg/tsdb/prometheus/prometheus.go

```diff
@@ -81,6 +90,36 @@ func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest)
 	return i.buffered.ExecuteTimeSeriesQuery(ctx, req)
 }
 
+func (s *Service) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
+	i, err := s.getInstance(req.PluginContext)
+	if err != nil {
+		return err
+	}
+
+	statusCode, bytes, err := i.resource.Execute(ctx, req)
+	if statusCode >= 300 {
+		safeErr := errors.New("no error specified")
```

Comment:
I am bit unsure about this error handling. In the resource.Execute we return status code even in cases where we don't do any request. Wouldn't it make sense to just return the error in such cases and let grafana handle it as internal errors? Seems like here we could leak random information by just sending all errors to front end.